### PR TITLE
feat: add duration to task and session

### LINKED ISSUE #208: Add duration to tasks and sessions in order to enable filtering

### PR DESCRIPTION
**Pull request by esoubiran-aneo**
fix #208

Sorting will be added with #207

### CODE REVIEW

File: Protos/V1/tasks_common.proto [hunk 1]
```diff
@@ -35,7 +35,9 @@ message TaskRaw {
   google.protobuf.Timestamp submitted_at = 12; /** The task submission date. */
   google.protobuf.Timestamp started_at = 13; /** The task start date. */
   google.protobuf.Timestamp ended_at = 14; /** The task end date. Also used when task failed. */
-  google.protobuf.Timestamp pod_ttl = 15; /** The pod TTL (Time To Live). */
```
Comment (lemaitre-aneo) [inline Protos/V1/tasks_common.proto:38]: Do not change the number of this field. It is a breaking change to do so.

File: Protos/V1/sessions_common.proto [hunk 2]
```diff
@@ -24,6 +24,7 @@ message SessionRaw {
 
   google.protobuf.Timestamp created_at = 5; /** The creation date. */
   google.protobuf.Timestamp cancelled_at = 6; /** The cancellation date. Only set when status is 'cancelled'. */
+  google.protobuf.Timestamp duration = 7; /** The duration. Only set when status is 'cancelled'. */
```
Comment (lemaitre-aneo) [inline Protos/V1/sessions_common.proto:27]: A duration is not a timestamp. You should use the proper type `Duration`.

**Review by lemaitre-aneo — COMMENTED** [review]
What is the use of those fields? I believe the timestamps are enough.

**Comment by aneojgurhem** [comment]
@esoubiran-aneo need some updates

**Review by esoubiran-aneo — COMMENTED** [review]
fix: update field to a duration type and rework values

File: Protos/V1/tasks_common.proto [hunk 3]
```diff
@@ -67,6 +70,8 @@ message TaskSummary {
   google.protobuf.Timestamp created_at = 5; /** The task creation date. */
   google.protobuf.Timestamp started_at = 6; /** The task start date. */
   google.protobuf.Timestamp ended_at = 7; /** The task end date. Also used when task failed. */
+  google.protobuf.Duration duration = 15; /** The task duration. Between the creation date and the end date. */
+  google.protobuf.Duration calculation_duration = 16; /** The task calculated duration. Between the start date and the end date. */
```
Comment (aneojgurhem) [inline Protos/V1/tasks_common.proto:74]: I think we can improve the two names !

Comment (aneojgurhem) [reply Protos/V1/tasks_common.proto:74]: `start_to_end_duration`
`processing_to_end_duration`
?

Comment (aneojgurhem) [reply Protos/V1/tasks_common.proto:74]: @esoubiran-aneo

Comment (esoubiran-aneo) [reply Protos/V1/tasks_common.proto]: Yep, updated

File: Protos/V1/tasks_common.proto [hunk 4]
```diff
@@ -35,6 +36,8 @@ message TaskRaw {
   google.protobuf.Timestamp submitted_at = 12; /** The task submission date. */
   google.protobuf.Timestamp started_at = 13; /** The task start date. */
   google.protobuf.Timestamp ended_at = 14; /** The task end date. Also used when task failed. */
+  google.protobuf.Duration duration = 20; /** The task duration. Between the creation date and the end date. */
```
Comment (aneojgurhem) [inline Protos/V1/tasks_common.proto:39]: @esoubiran-aneo and those too^^

**Pull request merged**